### DISCUSSION TITLE
Feature Request: Artisan command to clear logs

### DISCUSSION THREAD

**Discussion by JunaidQadirB** [laravel/framework#33658] [General]
Clearing log files seems like a chore when done manually. An artisan command to clear outdated or all log files would be a useful addition in my opinion. I also think that I shouldn't be reaching out for a package to manage logs.

**Comment by netpok** [comment]
The underlying monolog library already handles this you can control by setting the days parameter in config/logging.php:
```php
        'daily' => [
            'driver' => 'daily',
            'path' => storage_path('logs/laravel.log'),
            'level' => 'debug',
            'days' => 14,
        ],
```

If you are dealing with `.1.gz` files and using forge, it's created by an external logrotate process on the server so monolog cannot remove the files, you can disable it by removing the `/etc/logrotate.d/{{sitename}}` file.

**Reply by JunaidQadirB** [reply]
But that's not on-demand. My suggestion is just like I can clear the config or caches, I should also be able to clear the logs.

**Comment by ahinkle** [comment]
Adds too much “fluff” to the framework; other frameworks don’t support a CLI command to clear logs. 

This should easily be accomplishable by creating a bash command to clear your logs across all of your applications.

**Reply by vincenzoraco** [reply]
But as this is a repetitive action which many of us do, doesn't make sense to have it out-of-the-box?

**Reply by netpok** [reply]
Because the logging is not part of the framework, it already allows for configuring the underlying library to allow cleanup of old logs. Implementing this in the framework would mean the framework needs to follow the changes in monolog instead of relying on the provided interface.

**Comment by xitox97** [comment]
@JunaidQadirB Hi, I create a small package to solve your needs. It is a good simple problem for me to learn create my own package. You can check it out here https://github.com/xitox97/artisan-clear  

Any feedback would be greatly appreciated 😁

**Reply by netpok** [reply]
A few notes:
- I think it would be better to include "log" in the package name unless you intend to support more things to clear in the future
- Also you should take the location of the log file from the config.
- Maybe supporting daily logs would be a great addition but that's just an idea.

**Reply by xitox97** [reply]
> A few notes:
> 
> * I think it would be better to include "log" in the package name unless you intend to support more things to clear in the future
> * Also you should take the location of the log file from the config.
> * Maybe supporting daily logs would be a great addition but that's just an idea.

I wanted to add command that can clear all (cache, route, log) in future. Thank you for the idea, will add it soon 😄

**Reply by JunaidQadirB** [reply]
I appreciate your effortl. Definitely I'm not going to add a package for that and write my own command like you did.
But my point is a built-in option would be useful.
